### PR TITLE
remove unnecessary loops

### PR DESCRIPTION
**Pull request by bendavies**
spotted some improvements that can be made in https://github.com/vimeo/psalm/commit/8202af957d044c4ff23842878b2a718f45d3c2c9.
This should perform better for large arrays.

### CODE REVIEW

**Comment by muglug** [comment]
I had it in my head that `array_merge(...$arr)` only worked in PHP 7.4, glad that it's all good. Thanks!

**Comment by bendavies** [comment]
FYI the `array_values` is needed because `array_merge(...$arr)` doesn't work if `$arr` has string indexes.

File: src/Psalm/ErrorBaseline.php [hunk 1]
```diff
@@ -198,16 +200,11 @@ public static function update(
      */
     private static function countIssueTypesByFile(array $issues): array
     {
-        $normalized_data = [];
-
-        foreach ($issues as $file_issues) {
-            foreach ($file_issues as $issue_data) {
-                $normalized_data[] = $issue_data;
-            }
+        if (0 === count($issues)) {
```
Comment (snapshotpl) [inline src/Psalm/ErrorBaseline.php]: `[] === $issues` is "much" faster than `count`

Comment (muglug) [reply src/Psalm/ErrorBaseline.php]: updated in 947765c

Comment (bendavies) [reply src/Psalm/ErrorBaseline.php]: good spot. strangely that i knew that and already do it in all my code, but didn't here! thanks!

**Comment by someniatko** [comment]
@muglug prior to PHP 7.4, `array_merge(...$arr)` will fail if $arr is empty array.

**Comment by someniatko** [comment]
i used to workaround it as `array_merge([], ...$arr)`

**Pull request merged**